### DISCUSSION TITLE
request for feedback: draft of revised mission statement for the Eclipse IDE Working Group

### DISCUSSION THREAD

**Discussion by martinlippert** [eclipse-platform/eclipse.platform.releng.aggregator#2237] [General]
The Eclipse IDE WG is currently working on a revised mission statement and is looking for feedback from the involved projects, committers, and contributors. Once the mission statement is finalized, the goal is to have it on the working group website and as an entry point for potential new companies (who might be interested in joining the working group).

Important to note here is that this is not meant to be a mission statement for the projects themselves. It is the mission for the working group and their member companies in support of the projects involved.

Here is the current draft. Is it organized around three main sentences, to be short and crisp around what we are working towards as a working group.

---

#### Modernize the Platform and sustain its stability to provide a solid foundation for the next decade of building and maintaining IDEs and rich-client applications.

This is done by continuously modernizing the core parts of the platform, updating the code base to the latest Java versions and by implementing a slick, modern, and highly performant version of SWT as the UI layer.

#### Modernize the IDE Platform to provide a best in class platform to implement and ship custom IDEs for all kinds of languages.

This includes bringing the language server protocol integration to the next level as the central component of this platform, thus enabling users of the IDE platform to easily integrate existing language servers while extending functionality beyond what LSP supports at the same time.

#### Provide Best in Class Java Tooling on top of the IDE platform.

This includes using and continuously improving JDT to support the latest Java versions and running the Java tooling as a language server on top of the IDE platform. The Java tooling will not reduce or limit, but go beyond what is currently possible for Java in Eclipse. It will also serve as an example of how other languages and tools can be built on top of the IDE platform.

---

Looking forward to your feedback, comments, and suggestions.

**Comment by tfroment** [comment]
Thank you @martinlippert, hope to get some feedback from the community. It may be important to mention that, for me, the motivation behind these three sentences is indeed to define a set of "great goals" that can engage and motivate the community and potential contributors to invest in the IDE. I understand them this way, and I think they are interesting and motivating as currently written.

**Comment by mickaelistria** [comment]
My 2c (and really not worth more, feel free to ignore), against the overusage of "modernize"
"Modernize" has no concrete meaning, it's open to interpretations, trends, subjectivity; it's extremely mutable and there is no way to evaluate some "completeness" or even often to agree on the path of "progress"... It's something that is not so well-defined enough to make it a goal IMO.
Also, I find it too prominent: I know many RCP adopters who do not care about the theme, the colors, and who are much more interested in compatibility, stability, performance... in their toolchain (that can include older OSs). They're not looking for something modern, but for something efficient and sustainable on long-term; which have been the biggest quality of Eclipse Platform/IDE for a long time. So by putting a strong accent on "modernize", it seems like this WG is currently lowering priority on the compatibility/stability/performance aspect in favor of "modernize". If I were an adopter of Eclipse who got convinced by compatibility/stability/performance, I would be worried about this missing statement; and would either join the WG to influence it the other way round (optimistic) or take this statement seriously and embrace the change and evaluate further whether alternative platforms have bigger focus on compatiblity/stability/performance than what the Eclipse ecosystem is now affirming through this WG statement.

**Reply by martinlippert** [reply]
Thanks a lot @mickaelistria for taking a look and for providing your feedback, very much appreciated and well received. And I think you are emphasizing a very good point, I will try to bring this back to the working group and incorporate this aspect into the mission. From my point of view, the compatibility/stability/performance angle is still as important as the "modernize" part and I think the WG thinks the same way. But a new mission statement should not create a different impression, so thanks a lot for pointing that out. Again, very much appreciated.

**Reply by HannesWell** [reply]
> "Modernize" has no concrete meaning, it's open to interpretations, trends, subjectivity; it's extremely mutable and there is no way to evaluate some "completeness" or even often to agree on the path of "progress"... It's something that is not so well-defined enough to make it a goal IMO.

That's right, but I think it's also ok to have that goal that has no specific end, because I think the WG is not intended to have a specific end just like the Eclipse IDE will probably never be finished, doesn't it?
It's also right that modern is not even well defined in each point in time and open for interpretations, but again I think that can also be fine if the goal is to allow everyone interested to project their interpretation of modern (as long as this doesn't conflict with other views, it would just widen the 'modern' scope).



> it seems like this WG is currently lowering priority on the compatibility/stability/performance aspect in favor of "modernize". If I were an adopter of Eclipse who got convinced by compatibility/stability/performance, I would be worried about this missing statement;

I agree that stability should also be part of the statement since it is important as you explained. But I also I think that it should somehow emphasized that it doesn't mean stagnation and that stability might not last forever in every aspect. Of course Eclipse should not be disruptive all the time, but it might happen that some things are faded out with a clear migration path and sufficient time.
But I don't know if that can be put clearly in a short and snappy statement.

**Comment by opcoach** [comment]
As part of Eclipse's modernisation and image, it would be important to consider the **environmental aspect**. 

Coding, building, deploying in a green way is important today but nothing is really done for that. We are still using 20-25 years old technologies which were not in this scope... but we have to think about it ... not only to be more respectful of the environment, but also to become an open source reference in this field.  

More and more environmental standards are going to be introduced and will force industries to be more respectful of the environment... we're thinking of course of chemicals and transport, but the software industry is also concerned... 

Today, downloading 500 MB just to develop java code is still huge, especially when you consider that millions of developers are downloading... 

This area of development would bring a big number of advantages: 

- a clean image (eclipse is already free, so if it's clean it's even better)
- becoming a benchmark in this field, which will attract both developers and companies
- boost the Eclipse RCP aspect: a company that has to comply with or is concerned about environmental standards could get a label indirectly by using the RCP platform as a sober and clean software platform. 
- run on machines that consumes less energy (double effect)
- perhaps find funding to achieve these objectives.
- review the development criteria and modernise them
- increase the communication on Eclipse about this theme and boots its image
- enhance build processes that will cost less if we are using third parties partners like GitHub or gitlab 
- may be create a new working group about 'green development'
- ...

**Reply by martinlippert** [reply]
As you might know, I am personally quite active in the area of what sustainability (and green it) means for software engineering, so I sympathize a lot with your point. Thanks a lot for bringing this up.

Incorporating this into an updated mission statement for the working group would also mean to put some concrete ideas and activities behind this - which would be awesome. But it also adds a completely new and additional huge space and perspective to the working group. I am personally not sure if this would be a little overwhelming for the working group, given the limited number of involved parties resources that we have. Nevertheless, let's bring this topic back to the working group. Thanks again for raising this!
